riptide-httpclient/src/main/java/org/zalando/riptide/httpclient/RestAsyncClientHttpResponse.java

```diff
@@ -34,11 +34,13 @@ public String getStatusText() throws IOException {
 
     @Override
     public InputStream getBody() throws IOException {
-        return new FilterInputStream(response.getBody()) {
+        final InputStream body = response.getBody();
+        return new FilterInputStream(body) {
             @Override
             public void close() throws IOException {
-                if (in instanceof ConnectionReleaseTrigger) {
-                    ((ConnectionReleaseTrigger) in).abortConnection();
+                if (body instanceof ConnectionReleaseTrigger) {
+                    // effectively releasing the connection back to the pool in order to prevent starvation
+                    ConnectionReleaseTrigger.class.cast(body).abortConnection();
```

Comment:
We need abort for streaming responses, e.g. from Nakadi, where we can't read the remaining response body.